OsmAnd/src/net/osmand/plus/views/mapwidgets/MapMarkersWidgetsFactory.java

```diff
@@ -189,9 +190,16 @@ public void updateInfo(LatLon customLocation, int zoom) {
 			Location l = map.getMapViewTrackingUtilities().getMyLocation();
 			if (l != null) {
 				loc = new LatLon(l.getLatitude(), l.getLongitude());
-				lastKnownPosition = loc;
+				lastKnownPosition = l;
+			} else if (lastKnownPosition == null) {
+				loc = null;
 			} else {
-				loc = lastKnownPosition;
+				if (System.currentTimeMillis() - lastKnownPosition.getTime() < OsmAndLocationProvider.LOCATION_SHELF_LIFE) {
```

Comment:
May be it should be generical functionality that would cover all cases not only 1 widget?